src/components/pagination/Pagination.tsx

```diff
@@ -0,0 +1,159 @@
+import { useMemo } from "react";
+import { BUTTON_KIND, BUTTON_SIZE, Button, ButtonProps } from "../button";
+import { ChevronLeftIcon, ChevronRightIcon } from "../icons";
+import { PageElementProps, PaginationProps } from "./types";
+import { useStyletron } from "baseui";
+
+const PageElement = ({ page, linkMapper, pageHandler, label, disabled, active, buttonSize }: PageElementProps) => {
+  const buttonProps: ButtonProps = {
+    disabled,
+    kind: active ? BUTTON_KIND.primary : BUTTON_KIND.secondary,
+    size: buttonSize,
+  };
+  if (pageHandler) {
+    buttonProps.onClick = (e) => {
+      e.preventDefault();
+      pageHandler(page);
+    };
+  }
+  if (linkMapper && !disabled) {
+    return (
+      <li>
+        <a
+          href={linkMapper(page)}
+          onClick={(e) => {
+            e.preventDefault();
+            if (pageHandler) pageHandler(page);
```

Comment:
Why do we need additional pageHandler here, though we already have one in `buttonProps`?